### PR TITLE
Initialize redis on connection pool

### PR DESCRIPTION
**Pull request by lorenzograndi4**
Fixes https://travis-ci.com/github/WeTransfer/spaceship/builds/176169747

### CODE REVIEW

**Comment by lorenzograndi4** [comment]
I only added one test atm. The rest of the throttling tests I simply wrapped in a `context` for clarity. Maybe there should be one more test for the initialization of the leakybucket class, I'll add that

**Comment by julik** [comment]
Ah I see now! But should we create that much diff just for adding one context? Contexts imply that there will be multiple where different cases are covered, and here it is not the case.

The reason I am being so picky about it is that Prorate is somewhat complex to read/understand. Any extra stuff we add in tests contributes to that complexity.

File: spec/throttle_spec.rb [hunk 1]
```diff
@@ -21,9 +22,24 @@
 
   describe '#throttle!' do
     let(:throttle_name) { 'leecher-%s' % SecureRandom.hex(4) }
+    let(:r) { Redis.new }
+
+    context 'with a connection pool' do
+      it 'throttles and raises an exception' do
+        pool = ConnectionPool.new(size: 3, timeout: 3) { Redis.new }
```
Comment (dennisvdvliet) [inline spec/throttle_spec.rb:29]: you can use the `r` you set on line 25 here.

**Comment by julik** [comment]
👍

**Pull request merged**